### ISSUE TITLE
Add an option to disable tree-shaking?

### ISSUE THREAD

**Issue by ochafik** [rollup/rollup#505]
This would allow to work around bugs like #504.

**Comment by xinchaobeta** [comment]
+1

It is also useful for debugging code.

**Comment by tyler-johnson** [comment]
+1

I've already spent a few hours today working around issues with rollup removing function but not removing all the references to those functions. Also, if rollup is used to build tests for code coverage, all of the library's code should be in there.

**Comment by Ventajou** [comment]
I'd love such an option as well, I'm using Angular 1.x and because of the dependency injection system Rollup is removing my services

**Comment by eventualbuddha** [comment]
@Ventajou can you provide an example?

**Comment by Ventajou** [comment]
@eventualbuddha I just pushed an example to a repo:
https://github.com/Ventajou/rollup-treeshaking
The hack to get it to work is that line:
https://github.com/Ventajou/rollup-treeshaking/blob/master/src/controller.ts#L3

Hopefully it's clear enough to figure out...

**Comment by Rich-Harris** [comment]
`treeshake: false`/`--no-treeshake` is available in 0.26.1.

@Ventajou thanks for the repro, will look into it

**Comment by Rich-Harris** [comment]
@Ventajou so I've just had a quick poke, and it looks like Rollup is behaving as expected but rollup-plugin-typescript (or Typescript itself – not sure) isn't. I intercepted the output of the plugin to see what JavaScript was being generated, and for `controller.ts`...

```typescript
import MyService from './service'
// Uncomment the line below to "fix" the issue
// var foo = MyService;

export default class MyController {
  hello: string;
  constructor($myService: MyService) {
    this.hello = $myService.hello();
  }
}
```

...here's what I got:

```js
// Uncomment the line below to "fix" the issue
// var foo = MyService;
var MyController = (function () {
    function MyController($myService) {
        this.hello = $myService.hello();
    }
    return MyController;
}());
export { MyController as default };
```

In other words, the relevant `import` statement is getting stripped out, possibly because `MyService` isn't actually used anywhere in the file? (The type annotation doesn't count.) I don't know really know much about TypeScript, but it sounds like you should maybe raise an issue on [rollup-plugin-typescript](https://github.com/rollup/rollup-plugin-typescript).
